src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+
+    private int size;
+
+    public ArrayList() {
+        this.values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        increase();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkAddIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            increase();
+            T[] valuesTemp = (T[]) new Object[size - index + 1];
+            for (int i = index; i <= size; i++) {
+                valuesTemp[i - index] = values[i];
+            }
+            values[index] = value;
+            for (int i = index + 1; i <= size; i++) {
+                values[i] = valuesTemp[i - index - 1];
+            }
+            size++;
+        }
```

Comment:
```suggestion
        } else {
            increase();
            System.arraycopy(values, index, values, index + 1, size - index);
            values[index] = value;
            size++;
        }
```
better use System.arrayCopy, code will be smaller and no need to use two loops for that , i think this solution needs to work with your code(i'm still new so maybe did some mistake:( )

Reply:
I knew about this method, but as usual, I forget to read "Try to avoid these common mistakes while solving task". I thought it was better. )